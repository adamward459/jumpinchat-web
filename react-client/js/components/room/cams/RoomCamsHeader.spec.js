/* global window, it, beforeEach, describe */

import React from 'react';
import { shallow } from 'enzyme';
import RoomCamsHeader from './RoomCamsHeader.react';

describe('<RoomCamsHeader />', () => {
  let props;
  beforeEach(() => {
    props = {
      localStream: {
        stream: {
          getAudioTracks: jest.fn(() => ([])),
        },
      },
      canBroadcast: true,
      room: {
        name: 'room',
        users: [1, 2, 3, 4],
        settings: {
          description: 'foo',
        },
        attrs: {
          ageRestricted: false,
        },
      },
      playYoutubeVideos: true,
      feedCount: 3,
      userCount: 4,
      localAudioActive: false,
      showYoutubeVolume: false,
      showVolumeControl: false,
      broadcastRestricted: false,
    };
  });

  describe('broadcast button', () => {
    it('should show the broadcast button', () => {
      const wrapper = shallow(<RoomCamsHeader {...props} />);

      expect(wrapper.find('RoomBroadcastButton').length).toEqual(1);
    });
  });

  describe('local audio actions', () => {
    it('should show the actions if local stream has audio tracks', () => {
      props.localStream.stream.getAudioTracks = jest.fn(() => [{}]);
      const wrapper = shallow(<RoomCamsHeader {...props} />);
      expect(wrapper.find('RoomCamsLocalAudioActions').length).toEqual(1);
    });

    it('should not show the actions if no local stream', () => {
      props.localStream.stream.getAudioTracks = jest.fn(() => []);
      const wrapper = shallow(<RoomCamsHeader {...props} />);
      expect(wrapper.find('RoomCamsLocalAudioActions').length).toEqual(0);
    });
  });

  describe('room info', () => {
    it('should display correct room name', () => {
      const wrapper = shallow(<RoomCamsHeader {...props} />);
      expect(wrapper.find('.cams__RoomName').text()).toEqual('room');
    });

    it('should display correct cam count', () => {
      const wrapper = shallow(<RoomCamsHeader {...props} />);
      expect(wrapper.find('.cams__StreamCount').text()).toEqual('3');
    });

    it('should show correct user count', () => {
      const wrapper = shallow(<RoomCamsHeader {...props} />);
      expect(wrapper.find('.cams__ViewerCount').text()).toEqual('4');
    });

    it('should show display pic if one has been set', () => {
      props.room.settings.display = 'foo';
      const wrapper = shallow(<RoomCamsHeader {...props} />);
      expect(wrapper.find('.cams__RoomDisplayPic').length).toEqual(1);
    });

    it('should show room description if one exists', () => {
      props.room.settings.description = 'foo';
      const wrapper = shallow(<RoomCamsHeader {...props} />);
      expect(wrapper.getElement()).toMatchSnapshot();
    });

    it('should not show room description if none exists', () => {
      props.room.settings.description = null;
      const wrapper = shallow(<RoomCamsHeader {...props} />);
      expect(wrapper.getElement()).toMatchSnapshot();
    });

    it('should show topic if topic set', () => {
      props.room.settings.topic = {
        text: 'such topic',
      };
      const wrapper = shallow(<RoomCamsHeader {...props} />);
      expect(wrapper.getElement()).toMatchSnapshot();
    });

    it('should show topic instead of description if set', () => {
      props.room.settings.description = 'a description';
      props.room.settings.topic = {
        text: 'such topic',
      };
      const wrapper = shallow(<RoomCamsHeader {...props} />);
      expect(wrapper.getElement()).toMatchSnapshot();
    });
  });
});
