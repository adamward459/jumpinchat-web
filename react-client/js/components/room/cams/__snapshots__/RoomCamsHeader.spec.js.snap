// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RoomCamsHeader /> room info should not show room description if none exists 1`] = `
<div
  className="cams__Header"
>
  <div
    className="cams__RoomInfo"
  >
    <h2
      className="cams__RoomName"
    >
      room
    </h2>
    <div
      className="cams__InfoWrapper cams__StreamCountWrapper"
    >
      <span
        className="cams__InfoCount cams__StreamCount"
      >
        3
      </span>
      <span
        className="cams__InfoLabel cams__StreamCountLabel"
      >
        cams
      </span>
    </div>
    <div
      className="cams__InfoWrapper cams__ViewerCountWrapper"
    >
      <span
        className="cams__InfoCount cams__ViewerCount"
      >
        4
      </span>
      <span
        className="cams__InfoLabel cams__ViewerCountLabel"
      >
        viewers
      </span>
    </div>
    <RoomRestrictions
      restrictions={
        Object {
          "ageRestricted": false,
          "forceUser": undefined,
          "public": undefined,
          "requiresPassword": undefined,
          "requiresVerifiedEmail": undefined,
        }
      }
    />
  </div>
  <div
    className="cams__HeaderActions"
  >
    <RoomBroadcastButton
      canBroadcast={true}
      feedCount={3}
      localStream={
        Object {
          "stream": Object {
            "getAudioTracks": [MockFunction] {
              "calls": Array [
                Array [],
              ],
              "results": Array [
                Object {
                  "type": "return",
                  "value": Array [],
                },
              ],
            },
          },
        }
      }
      roomName="room"
    />
  </div>
</div>
`;

exports[`<RoomCamsHeader /> room info should show room description if one exists 1`] = `
<div
  className="cams__Header"
>
  <div
    className="cams__RoomInfo"
  >
    <h2
      className="cams__RoomName"
    >
      room
    </h2>
    <div
      className="cams__InfoWrapper cams__StreamCountWrapper"
    >
      <span
        className="cams__InfoCount cams__StreamCount"
      >
        3
      </span>
      <span
        className="cams__InfoLabel cams__StreamCountLabel"
      >
        cams
      </span>
    </div>
    <div
      className="cams__InfoWrapper cams__ViewerCountWrapper"
    >
      <span
        className="cams__InfoCount cams__ViewerCount"
      >
        4
      </span>
      <span
        className="cams__InfoLabel cams__ViewerCountLabel"
      >
        viewers
      </span>
    </div>
    <RoomRestrictions
      restrictions={
        Object {
          "ageRestricted": false,
          "forceUser": undefined,
          "public": undefined,
          "requiresPassword": undefined,
          "requiresVerifiedEmail": undefined,
        }
      }
    />
    <div
      className="cams__RoomDescriptionWrapper mobileHidden"
      title="foo"
    >
      <ClampLines
        buttons={false}
        className="cams__RoomDescription"
        delay={300}
        ellipsis="…"
        innerClassName="inner-card"
        lessText="Read less"
        lines={2}
        moreText="Read more"
        text="foo"
      />
    </div>
  </div>
  <div
    className="cams__HeaderActions"
  >
    <RoomBroadcastButton
      canBroadcast={true}
      feedCount={3}
      localStream={
        Object {
          "stream": Object {
            "getAudioTracks": [MockFunction] {
              "calls": Array [
                Array [],
              ],
              "results": Array [
                Object {
                  "type": "return",
                  "value": Array [],
                },
              ],
            },
          },
        }
      }
      roomName="room"
    />
  </div>
</div>
`;

exports[`<RoomCamsHeader /> room info should show topic if topic set 1`] = `
<div
  className="cams__Header"
>
  <div
    className="cams__RoomInfo"
  >
    <h2
      className="cams__RoomName"
    >
      room
    </h2>
    <div
      className="cams__InfoWrapper cams__StreamCountWrapper"
    >
      <span
        className="cams__InfoCount cams__StreamCount"
      >
        3
      </span>
      <span
        className="cams__InfoLabel cams__StreamCountLabel"
      >
        cams
      </span>
    </div>
    <div
      className="cams__InfoWrapper cams__ViewerCountWrapper"
    >
      <span
        className="cams__InfoCount cams__ViewerCount"
      >
        4
      </span>
      <span
        className="cams__InfoLabel cams__ViewerCountLabel"
      >
        viewers
      </span>
    </div>
    <RoomRestrictions
      restrictions={
        Object {
          "ageRestricted": false,
          "forceUser": undefined,
          "public": undefined,
          "requiresPassword": undefined,
          "requiresVerifiedEmail": undefined,
        }
      }
    />
    <div
      className="cams__RoomDescriptionWrapper mobileHidden"
      title="such topic"
    >
      <ClampLines
        buttons={false}
        className="cams__RoomDescription"
        delay={300}
        ellipsis="…"
        innerClassName="inner-card"
        lessText="Read less"
        lines={2}
        moreText="Read more"
        text="such topic"
      />
    </div>
  </div>
  <div
    className="cams__HeaderActions"
  >
    <RoomBroadcastButton
      canBroadcast={true}
      feedCount={3}
      localStream={
        Object {
          "stream": Object {
            "getAudioTracks": [MockFunction] {
              "calls": Array [
                Array [],
              ],
              "results": Array [
                Object {
                  "type": "return",
                  "value": Array [],
                },
              ],
            },
          },
        }
      }
      roomName="room"
    />
  </div>
</div>
`;

exports[`<RoomCamsHeader /> room info should show topic instead of description if set 1`] = `
<div
  className="cams__Header"
>
  <div
    className="cams__RoomInfo"
  >
    <h2
      className="cams__RoomName"
    >
      room
    </h2>
    <div
      className="cams__InfoWrapper cams__StreamCountWrapper"
    >
      <span
        className="cams__InfoCount cams__StreamCount"
      >
        3
      </span>
      <span
        className="cams__InfoLabel cams__StreamCountLabel"
      >
        cams
      </span>
    </div>
    <div
      className="cams__InfoWrapper cams__ViewerCountWrapper"
    >
      <span
        className="cams__InfoCount cams__ViewerCount"
      >
        4
      </span>
      <span
        className="cams__InfoLabel cams__ViewerCountLabel"
      >
        viewers
      </span>
    </div>
    <RoomRestrictions
      restrictions={
        Object {
          "ageRestricted": false,
          "forceUser": undefined,
          "public": undefined,
          "requiresPassword": undefined,
          "requiresVerifiedEmail": undefined,
        }
      }
    />
    <div
      className="cams__RoomDescriptionWrapper mobileHidden"
      title="such topic"
    >
      <ClampLines
        buttons={false}
        className="cams__RoomDescription"
        delay={300}
        ellipsis="…"
        innerClassName="inner-card"
        lessText="Read less"
        lines={2}
        moreText="Read more"
        text="such topic"
      />
    </div>
  </div>
  <div
    className="cams__HeaderActions"
  >
    <RoomBroadcastButton
      canBroadcast={true}
      feedCount={3}
      localStream={
        Object {
          "stream": Object {
            "getAudioTracks": [MockFunction] {
              "calls": Array [
                Array [],
              ],
              "results": Array [
                Object {
                  "type": "return",
                  "value": Array [],
                },
              ],
            },
          },
        }
      }
      roomName="room"
    />
  </div>
</div>
`;
